### PR TITLE
Refacto form error message into Component

### PR DESCRIPTION
**Pull request by MBausson**

### CODE REVIEW

File: app/views/servers/_form.html.erb [hunk 1]
```diff
@@ -1,22 +1,5 @@
...
-      </h4>
-
-      <ul>
-        <% @server.errors.full_messages.each do |msg| %>
-          <li><%= msg %></li>
-        <% end %>
-      </ul>
-    </div>
-  <% end %>
+<%= form_for(@server, html: { class: "form-horizontal", role: "form" }) do |f| %>
```
Comment (nicolas-brousse) [inline app/views/servers/_form.html.erb:1]: @MBausson not sure of this change, I think css class used should stay `form-server`. @B-Rass may you confirm this point?

File: config/locales/components.en.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+en:
+  form_errors_component:
+    title:
+      one: '1 error occured while saving :'
+      other: '%{count} errors occured while saving :'
```
Comment (nicolas-brousse) [inline config/locales/components.en.yml:5]:
```suggestion
      one: '1 error occured while saving:'
      other: '%{count} errors occured while saving:'
```

There is no space before `:` in English.

File: app/views/servers/_form.html.erb (hunk 1 above)
Comment (B-Rass) [reply app/views/servers/_form.html.erb:1]: Yes

File: app/components/form_errors_component.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      </button>
+
+      <h4><%= t(".title", count: @object.errors.count) %></h4>
+
+      <ul>
+        <% @object.errors.full_messages.each do |msg| %>
+          <li><%= msg %></li>
+        <% end %>
+      </ul>
+    </div>
```
Comment (B-Rass) [inline app/components/form_errors_component.rb:18]:
```suggestion
  erb_template <<~ERB
    <div class="col-md-12">
      <div class="alert alert-danger alert-dismissable" role="alert">
        <button type="button" class="close" data-dismiss="alert">
          <span aria-hidden="true">&times;</span>
          <span class="sr-only">Close</span>
        </button>

        <h4><%= t(".title", count: @object.errors.count) %></h4>

        <ul>
          <% @object.errors.full_messages.each do |msg| %>
            <li><%= msg %></li>
          <% end %>
        </ul>
      </div>
    </div>
```

File: app/views/card_types/index.html.erb [hunk 4]
```diff
@@ -40,48 +40,6 @@
...
-                            method: :delete,
-                            data: { confirm: t(".delete_confirmation") },
-                            class: 'btn btn-danger' %>
-              </td>
-            <% end %>
-          <% end %>
-          </tbody>
-        </table>
-      </div>
-
```
Comment (nicolas-brousse) [inline app/views/card_types/index.html.erb:84]: Sorry I forgot but we should keep this code. Do like in https://github.com/nanego/my-dcim/pull/129/files#diff-283e381c98f9ecd7e4299a56df5235ce565195923514c1aa4d26e21e970cc36e, be putting a false condition around for now.

**Review by B-Rass — CHANGES_REQUESTED** [review]

**Comment by B-Rass** [comment]
@MBausson I also had an error while creating an user and having a form error

**Comment by B-Rass** [comment]
> @MBausson I also had an error while creating an user and having a form error

Error I was talking about `Translation missing: fr.form.errors.title`

--

You have to use the component here

File: app/views/servers/_form.html.erb [hunk 5]
```diff
@@ -1,22 +1,5 @@
...
+  <%= render FormErrorsComponent.new(@server) %>
```
Comment (B-Rass) [inline app/views/servers/_form.html.erb:2]:
```suggestion
  <div class="col-md-12">
    <%= render FormErrorsComponent.new(@server) %>
  </div>
```

**Review by B-Rass — CHANGES_REQUESTED** [review]

File: app/views/servers/_modal_add_element_in_frame.html.erb [hunk 6]
```diff
@@ -14,18 +14,7 @@
...
-              </h4>
-
-              <ul>
-                <% @server.errors.full_messages.each do |msg| %>
-                  <li><%= msg %></li>
-                <% end %>
-              </ul>
-            </div>
+            <%= render FormErrorsComponent.new(@server) %>
           <% end %>
```
Comment (nicolas-brousse) [inline app/views/servers/_modal_add_element_in_frame.html.erb:18]: No needs of `if @server.try(:errors).try(:any?)`.

But by security, update `FormErrorsComponent#render?` to make it works if given object is `nil`.

File: app/views/servers/_form.html.erb (hunk 5 above)
Comment (nicolas-brousse) [reply app/views/servers/_form.html.erb]: @B-Rass it is one this file or for users?

**Review by B-Rass — APPROVED** [review]

**Review by nicolas-brousse — CHANGES_REQUESTED** [review]

File: app/views/servers/_form.html.erb (hunk 5 above)
Comment (nicolas-brousse) [reply app/views/servers/_form.html.erb]: this is good apparently

File: app/views/servers/_modal_add_element_in_frame.html.erb (hunk 6 above)
Comment (nicolas-brousse) [reply app/views/servers/_modal_add_element_in_frame.html.erb:18]: @MBausson I think this is the last change to apply, then we may be able to merge this work.

File: app/components/form_errors_component.rb [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  ERB
+
+  def initialize(object)
+    @object = object
+
+    super()
+  end
+
+  def render?
+    @object&.errors&.any?
```
Comment (nicolas-brousse) [inline app/components/form_errors_component.rb:28]: @MBausson I change this line

**Review by nicolas-brousse — APPROVED** [review]

**Pull request merged**